src/plots/cartesian/axes.js

```diff
@@ -82,21 +82,92 @@ axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption
     var refAttr = attr + 'ref';
     var attrDef = {};
 
-    if(!dflt) dflt = axlist[0] || extraOption;
+    if(!dflt) {
+        dflt = axlist[0] || (
+        ((extraOption === undefined) || (typeof(extraOption) === 'string')) ?
+            extraOption :
+            extraOption[0]
+        );
+    }
+    // so in the end, if dflt, axlist[], and extraOption were undefined, they
+    // are still undefined
     if(!extraOption) extraOption = dflt;
 
     // data-ref annotations are not supported in gl2d yet
 
+    var extraOptionAsList = extraOption;
+    if(extraOption) {
+        if(typeof(extraOption) === 'string') {
+            extraOptionAsList = [extraOption];
+        }
+    }
+
     attrDef[refAttr] = {
         valType: 'enumerated',
-        values: axlist.concat(extraOption ? [extraOption] : []),
+        values: axlist.concat(extraOptionAsList ? extraOptionAsList : []),
         dflt: dflt
     };
 
     // xref, yref
     return Lib.coerce(containerIn, containerOut, attrDef, refAttr);
 };
 
+/*
+ * An axis reference (e.g., the contents at the 'xref' key of an object) might
+ * have extra information appended. Extract the axis reference only.
+ *
+ * ar: the axis reference string
+ *
+ */
+axes.extractAxisFromAxisRef = function(ar) {
+    // TODO extractAxisFromAxisRef is more generic than this, this could be made
+    // more generic too
+    var mtch = (((ar !== undefined) && (typeof(ar) === 'string')) ?
+                ar.match(/^([xyz][0-9]*) domain/) :
+                undefined);
+    if(mtch) { return mtch[1]; }
+    return ar;
+};
+
+/*
+/*
+ * An axis reference (e.g., the contents at the 'xref' key of an object) might
+ * have extra information appended. Extract the extra information.
+ *
+ * ar: the axis reference string
+ *
+ */
+axes.extractInfoFromAxisRef = function(ar) {
```

Comment:
You refer to this as "info" or "opt" in the code but those are a bit too generic - perhaps this can be "reference type" (ie make variables like `xRefType` and this function can be called something like `getRefType`) and its return values could be `'paper'`, `'range'`, or `'domain'`?

Reply:
:+1: